scripts/core/kernel.yml

```diff
@@ -299,7 +311,6 @@ desc: "Set a Memory object as the argument value of a Kernel."
 class: $xKernel
 name: SetArgMemObj
 details:
-    - "The application must **not** call this function from simultaneous threads with the same kernel handle."
     - "The implementation of this function should be lock-free."
 params:
     - type: "$x_kernel_handle_t"
```

Comment:
Should we make the `hArgValue` optional here, this would be the equivalent of setting a buffer to null in OpenCL which I think is valid?

Reply:
Yeah I think that makes sense. Reading OpenCL it says `A NULL value can also be specified if the argument is a buffer object in which case a NULL value will be used as the value for the argument declared as a pointer to __global or __constant memory in the kernel.`. Although I'm not sure if UR has a concept of `__local`, `__global` etc..

Reply:
I've made this optional.